x-pack/plugins/cases/public/components/actions/tags/use_tags_action.tsx

```diff
@@ -37,16 +38,26 @@ export const useTagsAction = ({ onAction, onActionSuccess, isDisabled }: UseActi
     (tagsSelection: TagsSelectionState) => {
       onAction();
       onFlyoutClosed();
-      const casesToUpdate = selectedCasesToEditTags.map((theCase) => {
+
+      const casesToUpdate = selectedCasesToEditTags.reduce((acc, theCase) => {
         const tags = difference(theCase.tags, tagsSelection.unSelectedTags);
         const uniqueTags = new Set([...tags, ...tagsSelection.selectedTags]);
+        const uniqueTagsAsArray = Array.from(uniqueTags.values());
+
+        if (isEqual(theCase.tags, uniqueTagsAsArray)) {
```

Comment:
No need to update the case if the original tags are the same as the new ones

Reply:
How would sorting work for these arrays? If the arrays contain the same strings but are in a different order `isEqual` will return `false`. Maybe that's ok though, I'd have to check the backend to see if it would still throw an error in that situation. I suppose ideally we'd just skip the update since reordering the tags doesn't really do much in the UI.

Reply:
Thanks, I was not aware of that. I fixed it here https://github.com/elastic/kibana/pull/145608/commits/a200227eb3b7bc88d8352a76a0aef21f1594c999. As we have already a set of `uniqueTags` I iterate through the `theCase.tags` and check if it is a member of the `uniqueTags` st.